src/agent0/core/hyperdrive/agent/trade_result.py

```diff
@@ -44,6 +44,8 @@ class TradeResult:
     """If the trade failed due to slippage."""
     is_invalid_balance: bool = False
     """If the trade failed due to invalid balance."""
+    is_insufficient_approval: bool = False
```

Comment:
is this determined by the `check_for_insufficient_allowance` fn? why use "approval" here and "allowance" there?

Reply:
oops typo